packages/solidstart/src/vite/sourceMaps.ts

```diff
@@ -0,0 +1,56 @@
+import { sentryVitePlugin } from '@sentry/vite-plugin';
+import type { Plugin } from 'vite';
+import type { SourceMapsOptions } from './types';
+
+/**
+ * A Sentry plugin for SolidStart to enable source maps and use
+ * @sentry/vite-plugin to automatically upload source maps to Sentry.
+ * @param {SourceMapsOptions} options
+ */
+export function makeSourceMapsVitePlugin(options: SourceMapsOptions): Plugin[] {
+  return [
+    {
+      name: 'sentry-solidstart-source-maps',
+      apply: 'build',
+      enforce: 'post',
+      config(config) {
+        const sourceMapsPreviouslyNotEnabled = !config.build?.sourcemap;
+        if (options.debug && sourceMapsPreviouslyNotEnabled) {
+          // eslint-disable-next-line no-console
+          console.log('[Sentry SolidStart Plugin] Enabling source map generation');
+          if (!options.sourcemaps?.filesToDeleteAfterUpload) {
+            // eslint-disable-next-line no-console
+            console.warn(
+              `[Sentry SolidStart PLugin] We recommend setting the \`sourceMapsUploadOptions.sourcemaps.filesToDeleteAfterUpload\` option to clean up source maps after uploading.
+[Sentry SolidStart Plugin] Otherwise, source maps might be deployed to production, depending on your configuration`,
+            );
+          }
```

Comment:
l: Any reason to only log this when we detect that the user hasn't manually enabled sourcemaps?

Reply:
Aligning with the other packages, aside from that I do think it's nice to see why sourcemaps are being uploaded despite setting `sourcemap: false`.